### PR TITLE
Tests for updating a client with autoapprove scopes

### PR DESCRIPTION
**Pull request by willtran-**

### CODE REVIEW

File: common/src/test/java/org/cloudfoundry/identity/uaa/oauth/ClientAdminEndpointsTests.java [hunk 1]
```diff
@@ -694,4 +696,91 @@ public void testErrorHandler() throws Exception {
...
+    public void testUpdateClientWithAutoapproveScopesTrue() throws Exception {
+        List<String> scopes = Arrays.asList("foo.read","foo.write");
+        List<String> autoApproveScopes = Arrays.asList("true");
+        
+        input.setScope(scopes);
+        detail.setScope(scopes);
+        detail.setAutoApproveScopes(autoApproveScopes);
+        
+        Mockito.when(clientDetailsService.retrieve(input.getClientId())).thenReturn(
+                new BaseClientDetails(input));
```
Comment (fhanik) [inline common/src/test/java/org/cloudfoundry/identity/uaa/oauth/ClientAdminEndpointsTests.java]: This is faulty, if you look at the constructor, it doesn't copy the autoapprove field for you

	public BaseClientDetails(ClientDetails prototype) {
		this();
		setAccessTokenValiditySeconds(prototype.getAccessTokenValiditySeconds());
		setRefreshTokenValiditySeconds(prototype
				.getRefreshTokenValiditySeconds());
		setAuthorities(prototype.getAuthorities());
		setAuthorizedGrantTypes(prototype.getAuthorizedGrantTypes());
		setClientId(prototype.getClientId());
		setClientSecret(prototype.getClientSecret());
		setRegisteredRedirectUri(prototype.getRegisteredRedirectUri());
		setScope(prototype.getScope());
		setResourceIds(prototype.getResourceIds());
	}

Comment (willtran-) [inline common/src/test/java/org/cloudfoundry/identity/uaa/oauth/ClientAdminEndpointsTests.java]: input doesn't have autoapproved anything though, the test goes like: the existing client doesn't have autoapprove, update the existing client with autoapprove scopes, verify the autoapproved scopes are added.

Comment (fhanik) [inline common/src/test/java/org/cloudfoundry/identity/uaa/oauth/ClientAdminEndpointsTests.java]: I see that now. We don't transfer that field in here

https://github.com/cloudfoundry/uaa/blob/develop/common/src/main/java/org/cloudfoundry/identity/uaa/oauth/ClientAdminEndpoints.java#L588-588

**Comment by fhanik** [comment]
Superseding PR with new [feature branch](https://github.com/cloudfoundry/uaa/commit/db869b4084de1a04823759829512d7841221124a)

**Pull request closed without merging**